### PR TITLE
Add ressources for developpers

### PR DESCRIPTION
**Pull request by VincentRouvreau**
* default copyright files (Python and C++ versions)
* github guide for GUDHI contributors

### CODE REVIEW

**Comment by VincentRouvreau** [comment]
> Should they be linked from a contributing.md at the root (or whatever the "standard" name is)?

That's a good point.
Do you want me to do it on this PR ?
I have seen you have merged #182 we will have to update the document for submodule.
I can do it on this PR also.

**Comment by mglisse** [comment]
> Do you want me to do it on this PR ?

As you prefer.

> I have seen you have merged #182 we will have to update the document for submodule.

Yes.

File: .github/how_to_use_github_to_contribute_to_gudhi.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+This creates a directory gudhi-devel, which you are free to move around or rename. For the following, change to that directory:
+```bash
+cd gudhi-devel
+```
+
+## Submodule
+An interface to Hera for Wasserstein distance is available on an external git repository.
+Everytime you checkout master or merge from master, afterwards, you will need to run the command:
```
Comment (mglisse) [inline .github/how_to_use_github_to_contribute_to_gudhi.md]: Uh no, not that often. You only need to run this command once (while in a branch recent enough to have hera). Then you don't need to run it again until the version of hera we depend on changes, or we add more external dependencies. I think I would put this as a second line in "create a local clone", with a reminder near "git fetch".

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+For Python documentation, you can find it in the directories:
+* *src/python/doc* for the main page, installation instructionsand for the main pages of the modules
+* *src/python/gudhi/NAME_OF_THE_MODULE.pyx* for the documentation generated from the code.
+We use Sphinx to generate the code and you will be able to verify the result in CircleCI Sphinx target in the artifacts.
+
+## Something you want to improve in the code
+
+Please first take some time to read our [code conventions](code_conventions.md)
+
```
Comment (mglisse) [inline .github/CONTRIBUTING.md]: Should we add something to this file like "please check that your employer doesn't mind you contributing to open source. We don't ask for any paperwork but expect you don't submit anything you are not allowed to contribute."? (my formulation doesn't sound so good)

**Comment by mglisse** [comment]
This merge seems to have broken something:
> CMake Error: File /root/project/for_dev/for_maintainers/new_gudhi_version_creation.md does not exist.

**Comment by VincentRouvreau** [comment]
> This merge seems to have broken something:

Oops, my fault. I fixed it.

**Pull request merged**